src/serde/ser.rs

```diff
@@ -0,0 +1,991 @@
+use chrono::DateTime;
+use std::io::Cursor;
+
+use crate::ion_writer::IonWriter;
+use crate::result::IonFailure;
+use crate::serde::timestamp::ION_BINARY;
+use crate::types::Int;
+use crate::{
+    BinaryWriterBuilder, Element, IonError, IonResult, IonType, TextKind, TextWriterBuilder,
+    Timestamp,
+};
+use serde::ser::Impossible;
+use serde::{ser, Serialize};
+
+/// Serialize an object into pretty formatted Ion text
+pub fn to_pretty<T>(value: &T) -> IonResult<String>
+where
+    T: Serialize,
+{
+    ION_BINARY.with(move |cell| {
+        cell.set(true);
```

Comment:
Is `ION_BINARY` named correctly? Here you are setting it to `true` for Ion text.

Reply:
Renamed as this is used by both binary and text as a flag to indicate serde to treat the serialization or deserialization in terms of Ion format since timestamp and decimal representation in Ion differs from serde.